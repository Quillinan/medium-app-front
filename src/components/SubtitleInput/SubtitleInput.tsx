interface SubtitleInputProps {
  subtitle: string;
  onSubtitleChange: (value: string) => void;
}

const SubtitleInput: React.FC<SubtitleInputProps> = ({
  subtitle,
  onSubtitleChange,
}) => (
  <input
    data-testid='subtitle-input'
    type='text'
    placeholder='SubtÃ­tulo'
    value={subtitle}
    onChange={e => onSubtitleChange(e.target.value)}
    className='mb-2 border rounded p-2'
  />
);

export default SubtitleInput;
