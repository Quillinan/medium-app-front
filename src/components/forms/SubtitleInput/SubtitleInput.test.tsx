import { render, screen, fireEvent } from '@testing-library/react';
import SubtitleInput from './SubtitleInput';
import { useState } from 'react';

describe('SubtitleInput', () => {
  it('should render the component with the initial value', () => {
    render(
      <SubtitleInput subtitle='Initial Title' onSubtitleChange={vi.fn()} />
    );

    const input = screen.getByTestId('subtitle-input');
    expect(input).toBeInTheDocument();
    expect(input).toHaveValue('Initial Title');
  });

  it('should update the input value when typing', () => {
    const WrapperComponent = () => {
      const [subtitle, setSubtitle] = useState('');
      return (
        <SubtitleInput subtitle={subtitle} onSubtitleChange={setSubtitle} />
      );
    };

    render(<WrapperComponent />);

    const input = screen.getByTestId('subtitle-input');
    fireEvent.change(input, { target: { value: 'New Subtitle' } });

    expect(input).toHaveValue('New Subtitle');
  });

  it('should call onSubtitleChange with the correct value when changing the input', () => {
    const mockOnSubtitleChange = vi.fn();
    render(
      <SubtitleInput subtitle='' onSubtitleChange={mockOnSubtitleChange} />
    );

    const input = screen.getByTestId('subtitle-input');
    fireEvent.change(input, { target: { value: 'Change Test' } });

    expect(mockOnSubtitleChange).toHaveBeenCalledWith('Change Test');
  });

  it('should render the component correctly', () => {
    const { asFragment } = render(
      <SubtitleInput subtitle='' onSubtitleChange={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
