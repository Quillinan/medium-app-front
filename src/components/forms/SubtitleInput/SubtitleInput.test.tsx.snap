// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SubtitleInput > should render the component correctly 1`] = `
<DocumentFragment>
  <input
    class="mb-2 border rounded p-2"
    data-testid="subtitle-input"
    placeholder="SubtÃ­tulo"
    type="text"
    value=""
  />
</DocumentFragment>
`;
